subiquity/server/controllers/reboot.py

```diff
@@ -83,9 +84,12 @@ async def copy_logs_to_target(self, context):
     @with_context()
     def reboot(self, context):
         self.rebooting_event.set()
-        if self.opts.dry_run:
-            self.app.exit()
-        else:
-            if platform.machine() == 's390x':
-                run_command(["chreipl", "/target/boot"])
-            run_command(["/sbin/reboot"])
+
+        def f():
+            if self.opts.dry_run:
+                self.app.exit()
+            else:
+                if platform.machine() == 's390x':
+                    run_command(["chreipl", "/target/boot"])
+                run_command(["/sbin/reboot"])
+        self.app.aio_loop.call_soon(f)
```

Comment:
So this trick means that we can be sure that the POST() method returns before /sbin/reboot is invoked but it's not clear to me that we can be sure that the results of the POST() method are actually written to the socket the client is listening to before /sbin/reboot is invoked -- in fact I would be surprised if general aiohttp machinery *didn't* yield control to the event loop before any socket action happens and if it does, the event loop may well run `f` here before aiohttp machinery gets run again. We'd need to call reboot on some kind of signal that the response is done, but I don't think aiohttp provides such a signal.

However! What is this branch for again? :) The general idea is/was that you POST to /reboot and that blocks until reboot is invoked. So it seems to me the only case where what happens after it returns matters at all is dry-run mode, where the client should exit when the call returns. What does the desktop installer think it can do with the return value of this method? Do they just want to display a different screen before the reboot fully kicks in? Should we reboot a desktop install session in a more desktop environment friendly way?